File: main/src/model/meditateGoal.java
package model;

public class meditateGoal extends Goal{
    private final int goalAmount;
    private int currentAmount;
    private final String goalType = "Minutes";

    private meditateGoal(int goalAmount) {
        super();
        this.goalAmount = goalAmount;
    }

    public void setCurrentAmount(int currentAmount) {
        this.currentAmount = currentAmount;
    }

    public int getGoalAmount() {
        return goalAmount;
    }

    public int getCurrentAmount() {
        return currentAmount;
    }

    public String getGoalType() {
        return goalType;
    }
}

File: main/src_js/meditateGoal.js
import Goal from "./Goal";
export default class meditateGoal extends Goal {
    constructor(goalAmount) {
        super();
        this.goalAmount = 0;
        // private  currentAmount: number = 0;
        this.goalType = "Minutes";
        this.goalAmount = goalAmount;
    }
    //
    // setCurrentAmount(currentAmount: number) {
    //     this.currentAmount = currentAmount;
    // }
    setGoalAmount() {
        return this.goalAmount;
    }
    // getCurrentAmount(): number{
    //     return this.currentAmount;
    // }
    getGoalType() {
        return this.goalType;
    }
}